tests/core/test-mac.c

```diff
@@ -0,0 +1,132 @@
+/* Copyright 2014 Nodalink EURL
+ *
+ * This file is part of Packetgraph.
+ *
+ * Packetgraph is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published
+ * by the Free Software Foundation.
+ *
+ * Packetgraph is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Packetgraph.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+#include <memory.h>
+#include <unistd.h>
+
+#include <errno.h>
+#include <glib.h>
+#include <rte_mbuf.h>
+
+#include "utils/tests.h"
+#include "utils/mac.h"
+#include "tests.h"
+#include "utils/mempool.h"
+
+static void fill_addr(struct ether_addr *addr)
+{
+	addr->addr_bytes[0] = 0x01;
+	addr->addr_bytes[1] = 0x22;
+	addr->addr_bytes[2] = 0xF1;
+	addr->addr_bytes[3] = 0x07;
+	addr->addr_bytes[4] = 0xA1;
+	addr->addr_bytes[5] = 0x01;
+}
+
+static void test_print_mac(void)
+{
+	/* Build a MAC address for testing. */
+	struct ether_addr addr;
+
+	fill_addr(&addr);
+	/* Redirect stdout to buffer. */
+	char buffer[BUFSIZ] = {0};
+	int out_pipe[2];
+	int saved_stdout;
+	/* Save default stdout to recover later. */
+	saved_stdout = dup(STDOUT_FILENO);
+	/* Do the redirection. */
+	pipe(out_pipe);
+	dup2(out_pipe[1], STDOUT_FILENO);
```

Comment:
Bcause `dup()` only does a copy. So the error tested will shows up in the terminal while testing... That's why dup2: to prevent it. Because if II understood correctly we need clean test output.